apps/moodytunes/views.py

```diff
@@ -34,3 +43,30 @@ class EmotionPlaylistsView(FormView):
 
     def get_form_instance(self):
         return self.form_class(user=self.request.user)
+
+
+@method_decorator(login_required, name='dispatch')
+class SuggestSongView(FormView):
+    template_name = 'suggest.html'
+    form_class = SuggestSongForm
+
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(request.POST)
+
+        if form.is_valid():
+            code = form.cleaned_data['code']
+            fetch_song_from_spotify.delay(code)
+
+            logger.info(
+                'Added suggestion for song {} by user {}'.format(code, request.user.username),
+                extra={'fingerprint': 'added_suggested_song'}
+            )
+            messages.info(request, 'Your song has been slated to be added! Keep an eye out for it in the future')
+
+            return HttpResponseRedirect(reverse('moodytunes:suggest'))
+        else:
+            logger.warning(
+                'User {} suggested an invalid song code'.format(request.user.username),
```

Comment:
Add log of what code the user was requesting, this could help for debugging later to determine if the user simply entered a code for a song in our system or actually was sending bad data